### PR TITLE
W-16185857 upgrade pdk gr

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

**Review by JaatadiaMulesoft — DISMISSED** [review]

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 1]
```diff
@@ -0,0 +1,103 @@
+= Upgrading PDK
+ifndef::env-site,env-github[]
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc:2]: Do we need these attribute bits? I don't see them in most docs...

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+Flex Gateway Policy Development Kit (PDK) is composed of the following separate versioned components:
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: The [short description]([url](https://confluence.internal.salesforce.com/x/5LmeBw)) should cover what the topic is about, including user value.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+* Anypoint CLI PDK plugin
+* PDK Rust libraries
+* Anypoint Cargo plugin
+
+When you create a new custom policy project, the PDK Rust libraries and Anypoint Cargo plugin versions are the same as the Anypoint CLI PDK plugin used to create the project.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Is this topic about upgrading PDK or the custom policy project?

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+To upgrade a custom policy project created with an earlier version, you must upgrade your Anypoint CLI PDK plugin to the latest version. Then, you must modify the relevant files in your custom policy directory to upgrade the PDK Rust libraries and Anypoint Cargo plugin.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Maybe this is the short description?

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+[[anypoint-cli-pdk-plugin]]
+== Upgrade the Anypoint CLI PDK Plugin
+
+Your Anypoint CLI PDK plugin is used across all your PDK custom policies. When editing a custom policy created with an earlier plugin version, ensure your <<pdk-rust-libraries, PDK Rust libraries>> and <<anypoint-cargo-plugin, Anypoint Cargo plugin>> match your Anypoint CLI PDK plugin version.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Do you need the first sentence? It doesn't seem to add much.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+anypoint-cli-pdk-plugin 1.1.0
+├─ anypoint-cli-command 1.5.2
+│  └─ @oclif/plugin-help 5.1.12
+└─ @oclif/plugin-version 1.0.4
+----
+
+[[pdk-rust-libraries]]
+== Upgrade the PDK Rust Libraries
+
+The PDK library is the Rust Code library used by a specific custom policy. Different custom policy might use different PDK library versions.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Should this be "PDK Rust library" to match the heading?

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+To upgrade the PDK RUST code libraries, modify the `pdk` `dependency` and `pdk-test` `dev-dependency` versions in the `cargo.toml` file of the custom policy. The `pdk` and `pdk-test` versions must match and your Anypoint CLI PDK plugin must support the specified version.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Should RUST be "Rust" or "RUST" consistently?

Should code be "Code" or "code" consistently?

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc (hunk 6 above)
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]:
```suggestion
The PDK library is the Rust Code library used by a specific custom policy. Different custom policies might use different PDK library versions.
```

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+In the following example, both dependency versions are `1.0.0`. Modify it to your desired version:
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]:
```suggestion
In the following example, both dependency versions are `1.0.0`. Modify it to your required version:
```

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+pdk = { version = "1.0.0", registry = "anypoint" }
+
+[dev-dependencies]
+pdk-test = { version = "1.0.0", registry = "anypoint" }
+----
+
+[[anypoint-cargo-plugin]]
+== Upgrade the Anypoint Cargo Plugin
+
+The Anypoint Cargo plugin extends the functionality of Rust's package manager Cargo for use with the PDK Rust Libraries.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Avoid anthropomorphizing (with possessives).

Comment (hannanelson) [reply pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Oh, Cargo is the Rust package manager....

Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]:
```suggestion
The Anypoint Cargo plugin extends the functionality of Cargo, the Rust package manager, for use with the PDK Rust Libraries.
```

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+To upgrade your Anypoint Cargo plugin:
+
+. Open the `Makefile` of your custom policy.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc:71]: Don't use the element name in this way. Either say "the `Makefile` file" or refer to it generically "the Makefile".

(LMK if this is unclear...)

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+. Find the `install-cargo-anypoint` target, for example:
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc:72]: (This is correct.)

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
++
+----
+install-cargo-anypoint:
+	cargo install cargo-anypoint@1.0.0 --registry anypoint --config .cargo/config.toml
+----
+
+. Modify the version after the `@` character with the desired version.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]:
```suggestion
. Change the version after the `@` character to the required version.
```

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+. Save the `Makefile`.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc:80]: Same comment as line 71.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc (hunk 10 above)
Comment (hannanelson) [reply pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc:71]: See the Style Ref [guidance](https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-UIelements:~:text=Do%20not%20use%20a%20UI%20label%20to%20replace%20normal%20sentence%20syntax%3A)

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc (hunk 1 above)
Comment (glenn-rodgers-sf) [reply pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc:2]: All pages in Flex have them. So, I'm not 100% if we need them, but it does seem to be at least what we use.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc (hunk 3 above)
Comment (glenn-rodgers-sf) [reply pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Both?

I tried to make it make more sense in my edits.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc (hunk 9 above)
Comment (glenn-rodgers-sf) [reply pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: I adjusted this. I think any user this far along knows what Cargo is. It's a rust standard and is explained in the prereqs.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 14]
```diff
@@ -0,0 +1,107 @@
+= Upgrading PDK
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+Flex Gateway Policy Development Kit (PDK) is composed of the following separate versioned components that must each be updated to update a custom policy project:
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: What about something like this? 

To update a custom policy project, you must update each of these separately versioned Flex Gateway Policy Development Kit (PDK) components:

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+* Anypoint CLI PDK plugin
+* PDK Rust libraries
+* Anypoint Cargo plugin
+
+All your PDK custom policies on your device use the same Anypoint CLI PDK plugin. When you create a new custom policy project, the PDK Rust libraries and Anypoint Cargo plugin versions are the same as the Anypoint CLI PDK plugin used to create the project. The PDK Rust libraries and Anypoint Cargo plugin version of an individual custom policy project are unique from other policy projects. When you upgrade your Anypoint CLI PDK plugin, you must manually upgrade PDK Rust libraries and Anypoint Cargo plugin to be compatible with your Anypoint CLI PDK plugin version.
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: I wonder if there's a way to streamline this - or put the important bit first. 

Can you lead with this:

> When you upgrade your Anypoint CLI PDK plugin, you must manually upgrade PDK Rust libraries and Anypoint Cargo plugin to be compatible with your Anypoint CLI PDK plugin version.

And then give the rest as the "reason why"? 

Also, this phrase is duplicated: "the PDK Rust libraries and Anypoint Cargo plugin version"
Any way to combine?

Maybe:
When you create a new custom policy project, the PDK Rust libraries and Anypoint Cargo plugin versions are the same as the Anypoint CLI PDK plugin used to create the project and are unique from other policy projects.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+. Change the version after the `@` character to the required version.
+. Save the `Makefile`.
+. Execute the `make setup` command:
++
+[source,ssh]
+----
+make setup
+----
++
+[NOTE]
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc]: Does this build correctly? When I look at the preview, the Note text is in code font. Suggest removing the ---- as they're not necessary.

File: pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+----
+Executing the `make setup` command for a certain policy installs that policy's Anypoint Cargo plugin version.
+----
+. Execute the following command to verify the correct version is installed:
+[source,ssh]
```
Comment (hannanelson) [inline pdk/1.1/modules/ROOT/pages/policies-pdk-upgrade-pdk.adoc:96]: You might need a `+` here to get proper indentation.

**Review by hannanelson — DISMISSED** [review]

**Review by hannanelson — APPROVED** [review]

**Pull request merged**